js/views/AddressBarIndicators.js

```diff
@@ -0,0 +1,105 @@
+import _ from 'underscore';
+import loadTemplate from '../utils/loadTemplate';
+import baseVw from './baseVw';
+import { isMultihash } from '../utils';
+
+export default class extends baseVw {
+  constructor(options = {}) {
+    super(options);
+
+    this._state = {
+      hide: true,
+      url: '',
+      ...options.initialState || {},
+    };
+  }
+
+  updateVisibilityBasedOn(addressBarText) {
+    if (typeof addressBarText !== 'string') {
+      throw new Error('Please provide a valid address bar as string.');
+    }
+
+    const viewOnWebState = {
+      hide: true,
+      url: '',
+    };
+
+    const urlParts = this.getUrlParts(addressBarText);
+
+    if (urlParts.length > 1 && isMultihash(urlParts[0])) {
+      const supportedPages = ['store', 'home', 'followers', 'following'];
+      const currentPage = urlParts[1];
+
+      if (supportedPages.includes(currentPage)) {
+        const obDotCom = 'http://openbazaar.com';
+        const peerID = urlParts[0];
+
+        if (currentPage === 'store') {
+          // app: '/peerID/store/' => web: '/store/peerID/'
+          viewOnWebState.url = `${obDotCom}/store/${peerID}`;
+
+          if (urlParts.length === 3) {
+            // app: '/peerID/store/slug' => web: '/store/peerID/slug'
+            const slug = urlParts[2];
+            viewOnWebState.url = `${viewOnWebState.url}/${slug}`;
+          }
+        } else {
+          // app: '/peerID/(home|followers|following)' =>
+          // web: '/store/(home|followers|following)/peerID'
```

Comment:
This isn't commented out code. This documents the url in app versus the url translation for the web. I tried to make it concise since you guys don't seem to prefer much in-line documentation. Let me know what you prefer, but I'd prefer to keep my documentation in there to make it easier to understand when coming back to this. :)

Reply:
I see. The fat arrow threw me. We generally follow jsDoc format, though not that thoroughly, as you pointed out.

Reply:
@rmisio @jjeffryes : Let me know if those comment makes sense to you. Usually I would add a bit more language/verbosity to explain, but I tried to keep this concise for you guys. This comment is basically stating the form of the URL for each part of the block: What it looks like incoming (app) versus what it looks like as the URL I form for outgoing (web).